pulser-core/pulser/json/abstract_repr/validation.py

```diff
@@ -16,12 +16,12 @@
 from typing import Literal
 
 import jsonschema
+from referencing import Registry, Resource
 
-from pulser.json.abstract_repr import SCHEMAS, SCHEMAS_PATH
+from pulser.json.abstract_repr import SCHEMAS
 
-RESOLVER = jsonschema.validators.RefResolver(
-    base_uri=f"{SCHEMAS_PATH.resolve().as_uri()}/",
-    referrer=SCHEMAS["sequence"],
+REGISTRY: Registry = Registry().with_resources(
+    [("device-schema.json", Resource.from_contents(SCHEMAS["device"]))]
```

Comment:
Why is it device and not sequence ?

Reply:
Tbh, I think I never quite understood what the `referrer` argument did (in their docs, they have an example where they set it to `True` 🙃), I just went with something that worked. The schema being referred and the name (ie `device-schema.json`) was taken from the path in the `base_uri`.
Now, what we are doing is specifically building a `Registry` with the schemas that can be referred and the name they are referred by.

Reply:
Okay I just had a quick look at the doc but did not go further, I understood the role of the Registry and the Resource so let's go with this ;)